src/components/QuestionEdit/index.tsx

```diff
@@ -13,36 +13,27 @@ interface Props {
   questionNum: number;
   question: any;
   saveQuestion: any;
-  onQuestionChange: Function;
 }
 
-function QuestionEdit({ questionNum, question, saveQuestion, onQuestionChange }: Props, ref) {
-  const formDefaultValues = {
-    ...question,
-    options: question.options.length > 0 ? question.options : getEmptyOptions(2),
-  };
-  const { handleSubmit, setValue, watch, control } = useForm({ defaultValues: formDefaultValues });
+function QuestionEdit({ questionNum, question, saveQuestion }: Props, ref) {
+  const { handleSubmit, setValue, watch, control } = useForm({
+    defaultValues: {
+      ...question,
+      options: question.options.length > 0 ? question.options : getEmptyOptions(2),
+    },
+  });
   const { fields, append, remove } = useFieldArray({
     control,
     name: 'options',
   });
+  const options = watch('options');
   const [optionType, setOptionType] = useState<TabsValue>(
-    question.options.length === 1 && question.options[0].isCorrect ? 'withoutOptions' : 'withOptions',
+    options.length === 1 && options[0].isCorrect ? 'withoutOptions' : 'withOptions',
   );
+  const [focusOnLastOption, setFocusOnLastOption] = useState(false);
   const { showAlert } = useStore();
-  const { points, text, options, questionId } = watch();
   const isWithoutOptions = options.length === 1;
 
-  useEffect(() => {
-    onQuestionChange({
```

Comment:
We don't need to keep doing this. Just save the question when click the save button in the edit modal, don't need to save before that. This reduces some complexity.